src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -3702,3 +3702,41 @@ def show_bicep_cli_version(cmd):
 
 def list_bicep_cli_versions(cmd):
     return get_bicep_available_release_tags()
+
+def create_resourcemanager_privatelink(
+    cmd, resource_group, name, location):
+    rcf = _resource_privatelinks_client_factory(cmd.cli_ctx)
+    return rcf.resource_management_private_link.put(resource_group, name, location)
```

Comment:
Please note that the third parameter of method `rcf.resource_management_private_link.put` is not `location`, but `ResourceManagementPrivateLinkLocation`.  [code link](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/resources/azure-mgmt-resource/azure/mgmt/resource/privatelinks/v2020_05_01/operations/_operations.py#L632)
Therefore, you need to wrap `location` into `ResourceManagementPrivateLinkLocation` then passing the `ResourceManagementPrivateLinkLocation` into method `rcf.resource_management_private_link.put`
```suggestion
    cmd, resource_group, name, location):
    rcf = _resource_privatelinks_client_factory(cmd.cli_ctx)
    ResourceManagementPrivateLinkLocation = get_sdk(cmd.cli_ctx, ResourceType.MGMT_RESOURCE_PRIVATELINKS,
                                                    'ResourceManagementPrivateLinkLocation', mod='models')
    resource_management_private_link_location = ResourceManagementPrivateLinkLocation(location=location)
    return rcf.resource_management_private_link.put(resource_group, name, resource_management_private_link_location)
```

Reply:
```suggestion
    cmd, resource_group, name, location):
    rcf = _resource_privatelinks_client_factory(cmd.cli_ctx)
    ResourceManagementPrivateLinkLocation = get_sdk(cmd.cli_ctx, ResourceType.MGMT_RESOURCE_PRIVATELINKS,
                                                    'ResourceManagementPrivateLinkLocation', mod='models')
    resource_management_private_link_location = ResourceManagementPrivateLinkLocation(location=location)
    return rcf.resource_management_private_link.put(resource_group, name, resource_management_private_link_location)
```

Reply:
This is the problem that caused the CLI tests to fail